### DISCUSSION TITLE
[Feature Request]: Ability to hide the Dock App icon

### DISCUSSION THREAD

**Discussion by drewjoh** [beyondcode/herd-community#389] [Ideas]
## Feature Description

Please allow for an option to remove Herd from the dock.

I don't remember Herd living in the dock and menu bar prior to v1.4.

There's a few of reasons for this, and I know an issue was [raised and closed](https://github.com/beyondcode/herd-community/issues/328) on this topic already, but I would like to foster some more discussion on this topic.

My own issue, aside from there being an extra dock icon for a rarely accessed app, is that **Herd now always lives in my task switcher** (Cmd+Tab) as an application I have to navigate around.

Some may like a minimal menu bar, which is great.  I personally have a minimal dock.
**Herd breaks existing conventions that all other apps** I use on MacOS follow.

Herd is not the type of app that always has a window open.  Perhaps this is different with the new Pro features?  But just to have my Laravel apps running, I just need it to be running in the background.

If this convention does indeed change with Pro features (Herd is a window that is always opened and accessed), I think that's the argument for it needing to be optional.  😀 

Precedent for this are in apps like I'd like DBngin, Flux, Rectangle app, etc.

Many thanks

**Comment by mpociot** [comment]
Is Herd in the Dock for you the entire time? It should only have an icon when you open one of its windows (Dumps, Logs, Settings, etc. )
As soon as you close them, the icon gets hidden. 

Maybe you could share a video?

**Reply by drewjoh** [reply]
Great idea!  I'll attach a video here.

In making the video I think I found the "bug":  it starts up without the dock icon, when I open settings the dock icon shows up, when I close settings, the dock icon appears to be "stuck" and doesn't go away.

So an interim solution is I can quit and open Herd, but just not open the settings.  😀 

https://github.com/beyondcode/herd-community/assets/66233/af1387c6-5bcb-4dc3-bf4e-16080bf0e5a3

**Reply by drewjoh** [reply]
I'll add I think this is great behavior btw.  If there's a window available, show it in the dock so I can switch to it.  
If no windows exist, remove it from the dock.  😁

**Comment by RichieMcMullen** [comment]
Any updates on this? I have the same issue, herd remains in the dock after the window is closed.

**Comment by timrspratt** [comment]
Same issue for me. Icon remains with no windows open

**Comment by mpociot** [comment]
Yeah, I paired on this with James Brooks and apparently this occurs when you're using the "multiple desktop" feature in macOS, which makes the app believe it has multiple windows still open.
